### DISCUSSION TITLE
Microsoft Update expectations, do we upgrade 7.2 and 7.3 to 7.4?

### DISCUSSION THREAD

**Discussion by StevenBucher98** [PowerShell/PowerShell#21132] [General]
Hello everyone! I wanted to understand better the impact and or value of upgrading MSI users from 7.2 and 7.3 to 7.4? Is this something that would be helpful or more harmful for those upgrading pwsh? No changes are planned but would love to hear folks thoughts on this idea.

**Comment by rhubarb-geek-nz** [comment]
There are a few breaking changes, the three that spring to mind are change in default character sets, native libraries in modules and the JSON parser. You would need to go through the list and assess the pain of each breaking change.

**Comment by fMichaleczek** [comment]
![image](https://github.com/PowerShell/PowerShell/assets/6187529/0743f4e6-7e74-428c-a639-d30259ccb1e0)

imho, you should not upgrade software that is still supported.

**Reply by OranguTech** [reply]
I could see/might make the argument that older "stable" (7.3) should/would be expected to update to latest stable (7.4) (by default, unless overridden). But in-support  LTS (7.2) should not update to latest LTS (7.4) until explicitly enabled, or it goes out-of-support.

**Comment by kilasuit** [comment]
For management at scale this is something that I personally would like to be able to set as a new RegKey that should in future versions of the installer be able to be applied at install time, as well as post-install via GPO or other automatable methods & I think should allow for the following upgrade matrix

Current LTS -> Current LTS (until out of support & never update to latest Stable or Newest LTS) - examples being `7.2.x -> 7.2.x`  & `7.4.x -> 7.4.x`
Current LTS -> Current LTS -> Next Stable  - `7.2.x -> 7.2.x -> 7.3` or `7.4.x -> 7.4.x -> 7.5.x` (likely not going to be well used but should be an option for testing)
Current LTS -> Current LTS (until out of support) -> Next LTS - Missing any Stable versions  - `7.2.x -> 7.2.x -> 7.4.x` or `7.4.x -> 7.4.x -> 7.6.x`
Stable -> Stable (until out of support) -> LTS  - `7.3.x -> 7.3.x -> 7.4.x`
Stable -> Stable -> Next Stable - `7.3.x -> 7.3.x -> 7.5.x`

Especially with this point above, this matrix ensures that current version installed is the one that they want upgraded without an auto upgrade to a new version on a different development branch from what they are currently running
> imho, you should not upgrade software that is still supported.

**Reply by rhubarb-geek-nz** [reply]
Which installer are you referring to? Are you suggesting per-package registry configuration to be recognized by WSUS, WinGet, Windows App Store, chocolatey etc? Those are the components which decide on the updates.

**Reply by kilasuit** [reply]
As mentioned its the MSI installer package

No matter how you install the MSI if you opt-in to Microsoft Update you can move the update process away from those other tooling to be handled by Microsoft Update
Which for me is preferable in some of my environments

**Reply by rhubarb-geek-nz** [reply]
Yes, I am just not clear on the mechanics. An MSI contains a version and various database key/values in it, but it is not the component that decides which MSI is going to be installed/updated.
